README.md

```diff
@@ -104,7 +70,37 @@ case fetch_user(1) do
 end
 ```
 
-[Forum discussion on `with` naming](https://elixirforum.com/t/alternative-to-with-specific-to-result-tuples/3264)
+####  Error Matching
```

Comment:
I'm reworking the examples again to be more cohesive as a whole. Perhaps it will be better... :thinking: